Assets/AnkrSDK/Scripts/Core/Data/EventFilterRequest.cs

```diff
@@ -1,11 +1,91 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using Nethereum.ABI.FunctionEncoding.Attributes;
 using Nethereum.RPC.Eth.DTOs;
 
 namespace AnkrSDK.Core.Data
 {
 	public class EventFilterRequest<TEvDTO>
 	{
-		public TEvDTO request { get; set; }
-		public BlockParameter fromBlock { get; set; }
-		public BlockParameter toBlock { get; set; }
+		private class Topic
+		{
+			public string Name { get; set; }
+			public string Alias { get; set; }
+			public List<object> Values { get; set; }
+		}
+		
+		private readonly IEnumerable<Topic> _topics;
+		
+		public BlockParameter FromBlock { get; set; }
+		public BlockParameter ToBlock { get; set; }
+
+		public EventFilterRequest()
+		{
+			_topics = CollectTopics();
+		}
+
+		public void SetFromBlock(BlockParameter fromBlock)
```

Comment:
why do we need setter methods if we have public set for FromBlock/ToBlock properties?

Reply:
I thought that one way of setting variables will be better. I mean if we have AddTopic why we need to recomend set another variables directly.